apache/superset
# fix(elasticsearch): time_zone setting does not work for cast datetime expressions

### SUMMARY


The elasticsearch `CAST` function does not take effect for the `time zone` setting. In ES7.8 and above, we can use the `DATETIME_PARSE` function to solve this problem.

In the current master code, the `extra` dict in the database contains the `version` key. We can use version to solve this problem. According to the instructions, version is set for `presto` at the beginning, and ES should also reuse this field.


### TESTING INSTRUCTIONS


1. Install the latest elasticsearch-dbapi version, 0.2.6, which supports setting the time_zone parameter `pip install elasticsearch-dbapi==0.2.6`
2. Follow the steps in #16726 to create an ES data source, create a data source, set the time_zone parameter and version
<img width="530" alt="截屏2021-10-09 下午4 47 33" src="https://user-images.githubusercontent.com/40813941/136651378-32fe75c1-a703-416d-ae7c-cdf3db122bb4.png">

3. Follow the operation steps in #16726 to analyze, and you can see that the DATETIME_PARSE function and time_zone parameter play a correct role

<img width="1903" alt="截屏2021-10-09 下午4 50 14" src="https://user-images.githubusercontent.com/40813941/136651464-9e86c7b8-c2ae-4026-b09c-62f36aacfb02.png">

### ADDITIONAL INFORMATION


- [x] Has associated issue: #16726
